demo/api/src/app.module.ts

```diff
@@ -55,13 +56,22 @@ export class AppModule {
             imports: [
                 ConfigModule.forRoot(config),
                 DbModule,
-                GraphQLModule.forRootAsync({
+                GraphQLModule.forRootAsync<ApolloDriverConfig>({
                     driver: ApolloDriver,
                     imports: [BlocksModule],
                     useFactory: (dependencies: Record<string, unknown>) => ({
                         debug: config.debug,
                         playground: config.debug,
                         autoSchemaFile: "schema.gql",
+                        introspection: process.env.NODE_ENV === "development",
+                        formatError: (error) => {
+                            if (process.env.NODE_ENV !== "development") {
+                                if (error instanceof ValidationError) {
+                                    return new ValidationError("Invalid request.");
+                                }
+                            }
+                            return error;
+                        },
```

Comment:
It's not very nice that we need this lines in every of our projects. We could make it a little bit easier by providing the formatError function in the library. Something like this:

```ts
// Library
export const disableGraphQLSchemaSuggestions = {
  formatError: (error) => {
    if (process.env.NODE_ENV !== "development") {
      if (error instanceof ValidationError) {
        return new ValidationError("Invalid request.");
      }
    }
    return error;
  },
}

// Application
{
  useFactory: (dependencies: Record<string, unknown>) => ({
    ...disableGraphQLSchemaSuggestions
  }
}
```

@thomasdax98 what do you think?

Reply:
We could also think of extracting the whole GraphQL config to the library to open up the opportunity to make project wide changes in the future without the need of adjusting each project manually

Reply:
@johnnyomair Definitely better than pasting the code to each project

@jomunker I guess we could move some parts of the config to the library. It's probably unchanged in most of our projects.

```ts
{
    debug: config.debug,
    playground: config.debug,
    autoSchemaFile: "schema.gql",
    context: ({ req }: { req: Request }) => ({ ...req }),
    cors: {
        origin: config.corsAllowedOrigin,
        methods: ["GET", "POST"],
        credentials: false,
        exposedHeaders: [],
    },
    useGlobalPrefix: true,
    // See https://docs.nestjs.com/graphql/other-features#execute-enhancers-at-the-field-resolver-level
    fieldResolverEnhancers: ["guards", "interceptors", "filters"] as Enhancer[],
    buildSchemaOptions: {
        fieldMiddleware: [BlocksTransformerMiddlewareFactory.create(dependencies)],
    },
}
```

We need a deepmerge for nested overrides

Reply:
I'm not sure if it's a good idea to export the config in the library. We could also discuss having a custom GraphQLModule that should be used instead of the one provided by `@nestjs/graphql`. Similar to what we do with the MikroOrmModule: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/mikro-orm/mikro-orm.module.ts

Reply:
@nsams what do you think?

Reply:
imho don't overcomplicate this. Every standard nestjs application looks exactly like that.

We can consider creating a comet-upgrade script that does this modification.

Reply:
Good point! @johnnyomair @thomasdax98 so how should i continue?

Reply:
@jomunker let's just keep the additional option in the application code and add a upgrade script. Also, we should add the same change in Starter

Reply:
Starter PR: https://github.com/vivid-planet/comet-starter/pull/215